test/e2e/lib/broker.go

```diff
@@ -70,6 +70,13 @@ func (a BrokerMetricAssertion) Assert(client *monitoring.MetricClient) error {
 		if err != nil {
 			return fmt.Errorf("metric has invalid response code label: %v", ts.GetMetric())
 		}
+
+		// Workarounds to reduce test flakiness caused by sender pod retry sending events (which will cause unexpected response code).
```

Comment:
If I understand it correctly, the sender pod will retry if it receives non 202 code? I think the comment could clarify this.

I actually think this fix is desired even if #1058 is fixed. We are essentially saying that we only count for successful events, assuming the broker could return non 202 during tests.

Reply:
If my comment above is correct, I think we can just `continue` on line #81 instead of returning an error. We rely on successful event count to verify the test correctness.

Reply:
The sender pod will only retry for three statusCode, 404, 500 and 503. Maybe we can remove 500 if #1058 is fixed, as 500 here is mainly for reducing flakiness caused by WI credential sync (This might be fixed after #1058 lands)? And continue keep 403 and 503?

Reply:
It depends on what the expectations are but It doesn't seem like a bad idea for the sender pod to retry on non-202, even if we fix those issues. This is not a broker availability test anyways. But I am also OK with dealing with each code separately.

Reply:
Sounds good, I just updated the comments. Could you check if that is ok?